java/src/shared/model/game/Game.java

```diff
@@ -718,15 +731,17 @@ public void discardCards(int playerIndex, List<ResourceType> cards) throws Playe
     public void rollNumber(int value) throws Exception {
         //Is value a 7 - robber
         if(value == 7){
-            //Go to discarding phase before robbing if any player has to discard
-            getPlayers().forEach(player ->{
-                if(player.canDiscardCards()){
-                    turnTracker.setPhase(TurnTracker.Phase.DISCARDING);
-                    return;
+            needToDiscard.clear();
```

Comment:
Shouldn't the canDiscard function tell you whether or not a player needs to discard?